### PR TITLE
Disable node editing in review mode

### PR DESCRIPTION
**Pull request by my-tien**
Can select nodes or change active node.

Tree edits still possible, e.g. merging, moving nodes between trees.

### CODE REVIEW

**Comment by Optiligence** [comment]
>Tree edits still possible, e.g. merging, moving nodes between trees.
Intentional or just missing?

**Comment by my-tien** [comment]
Intentional, because there exists no `AnnotationMode::TreeEditing` atm.

**Comment by Optiligence** [comment]
is there a reason to distinguish between node and tree editing?

File: widgets/tools/skeletonview.cpp [hunk 1]
```diff
@@ -638,17 +639,18 @@ SkeletonView::SkeletonView(QWidget * const parent) : QWidget{parent}
...
         treeContextMenu.actions().at(i++)->setEnabled(selectedTrees.size() > 0);//restore default color
         treeContextMenu.actions().at(i++)->setVisible(selectedTrees.size() > 0 && containsMesh);//remove mesh
-        treeContextMenu.actions().at(deleteActionIndex = i++)->setEnabled(selectedTrees.size() > 0);//delete
+        treeContextMenu.actions().at(deleteActionIndex = i++)->setEnabled(nodeEditing && selectedTrees.size() > 0);//delete
         //display the context menu at pos in screen coordinates instead of widget coordinates of the content of the currently focused table
         treeContextMenu.exec(treeView.viewport()->mapToGlobal(pos));
         // make some actions always available when ctx menu isn’t shown
         treeContextMenu.actions().at(copyActionIndex)->setEnabled(true);
-        treeContextMenu.actions().at(deleteActionIndex)->setEnabled(true);
+        treeContextMenu.actions().at(deleteActionIndex)->setEnabled(nodeEditing);
```
Comment (Optiligence) [inline widgets/tools/skeletonview.cpp]: This is not right. The reason those are always enabled is that these are shortcuts and there is nothing set up for when the conditions, which specify when they are available (e.g. changed work mode), change. The logic specifying these conditions is therefore in the actions callback.

**Comment by my-tien** [comment]
Well, maybe you want to open a lot of files for review and then want to remove some without clearing the whole skeleton again. In these cases you’d want to be able to delete trees.

**Comment by Optiligence** [comment]
Maybe? Because i would currently put everything under `SkeletonEditing`, but we can also add `TreeEditing`.

**Comment by my-tien** [comment]
Maybe not maybe. I did that :P

**Pull request merged**